lewis/adapters/epics.py

```diff
@@ -321,7 +321,7 @@ def _create_getter(self, func, *targets):
             raise RuntimeError(
                 'The function \'{}\' does not look like a getter function. A valid getter '
                 'function has no arguments that do not have a default. The self-argument of '
-                'methods does not count towards that number.')
+                'methods does not count towards that number.'.format(final_callable.__name__))
```

Comment:
No, it doesn't work with stream either... modifying `dual_device` with:

```python
from functools import partial

def multiply(x, y):
    return x * y

class VerySimpleDevice(Device):
    param = 10
    _second = 2.0

    dbl = partial(multiply, 2)
    #def dbl(self, y):
    #    return 2 * y
```
and
```python
    commands = {
        Var('param', read_pattern=r'P\?$', write_pattern=r'P=(\d+)', argument_mappings=(int,),
            doc='An integer parameter.'),
        Cmd('dbl', scanf('D=%d'))
    }
```
Works with the commented out `def` version, but fails with the class version due to this in `Func.__init__`:
```python
        try:
            inspect.getcallargs(func, *[None] * self.matcher.arg_count)
        except TypeError:
            raise RuntimeError(
                'The number of arguments for function \'{}\' matched by pattern '
                '\'{}\' is not compatible with number of defined '
                'groups in pattern ({}).'.format(
                    getattr(func, '__name__', repr(func)), self.matcher.pattern,
                    self.matcher.arg_count
                ))
```
`getcallargs` throws an exception stating `func` is not a Python function, then our RuntimeError about argument number mismatch is displayed (since it was a TypeError).

This is certainly not a common use-case, but it's one of those things you might expect to work and be very disappointed when it doesn't. It's Python, after all, and with duck typing you'd expect to be able to use anything that's callable.

That said, there is a simple workaround: Just wrap your class call or partial in a real method or function.